Update to the latest available sqlite4java artifacts

georgi-neykov-hub opened the pull request:
### Overview
Add support for CTEs in `SQLiteDatabase`  by updating the underlying sqlite4java library

### Proposed Changes
- Update sqlite4java to version 1.0.392 bundled
with SQLite 3.8.7 which comes with support for CTEs
- Use the first-hand natives Maven  artifacts published by almworks.
- Update `SQLiteLibraryLoader` and related `SQLiteLibraryLoaderTest` for
the updated library paths and names
- Add unit test functions to assert CTE support
- The original repo of the project is at
https://bitbucket.org/almworks/sqlite4java

hoisie commented on shadows/framework/src/main/java/org/robolectric/shadows/util/SQLiteLibraryLoader.java:
```diff
@@ -113,17 +119,34 @@ private String getOsPrefix() {
+          case "x86_64":
+          case "amd64":
+            return "amd64";
+        }
```
please add a break statement after this to avoid fallthrough (ErrorProne complains about this).

hoisie commented on shadows/framework/src/main/java/org/robolectric/shadows/util/SQLiteLibraryLoader.java:
```diff
@@ -113,17 +119,34 @@ private String getOsPrefix() {
+            case "x86_64":
+            case "amd64":
+              return "x64";
+          }
```
same here

hoisie reviewed (commented):
Thanks for doing this, it looks great. I like the new resource directory structure. Sorry for taking so long to review.

Would you be able to run google-java-format on the results? I wrote up a quick wiki page about this:
https://github.com/robolectric/robolectric/wiki/Running-google-java-format

georgi-neykov-hub commented:
@hoisie , thank you for reviewing the pull request, honestly I thought it was never going to happen. 

* I've  added the `break`s in the nested `switch` statements and ran `google-java-format`. 
* There are some failing upstream tests unrelated to my changes.
* I updated my fork to be up-to-date with the current upstream `master` branch and finally squashed and rebased everything tas it was absurdly outdated. Hope it is ready for merging now.

BTW, there some [recent movement](https://bitbucket.org/almworks/sqlite4java/issues/89/upgrade-sqlite-to-the-latest-version-prior) in `sqlite4java`'s repository, the maintainers and contributors have been doing updates and there seems to be an effort to publish an updated version of the library with the latest available SQLite (3.30.0+) baked in.

Are there any stoppers or AOSP compatibility issues to have the latest SQLite engine shipped with robolectric? 
If not, I am considering to contact the repo owners and possibly help with the new release and bring the changes here. Can I count it will take less than 12 months to have the update reviewed?

hoisie commented:
Thanks for cleaning it up and rebasing. I am not 100% sure which version sqlite is used in Android R, but I believe that the Android platform makes an effort to keep it up to date, so it would be great to get the 3.30.0+ available in Robolectric.
